llvm/lib/Transforms/Utils/LoopUnroll.cpp

```diff
@@ -731,6 +740,16 @@ llvm::UnrollLoop(Loop *L, UnrollLoopOptions ULO, LoopInfo *LI,
           NewPHI->eraseFromParent();
         }
 
+        // Eliminate copies of the loop heart intrinsic, if any.
```

Comment:
If I understand the rules here correctly, this is only necessary if you're not completely unrolling the loop?

Reply:
Correct. If the loop is completely unrolled, then the loop intrinsic is redundant. Some other pass should eventually clean that up. But when a loop is partially unrolled, those copies of the heart violate the static rules, and all of them should be replaced by the instance from the header.